### PR TITLE
[stable17] Fix password changes in link and mail shares

### PR DESCRIPTION
**Pull request by danxuliu**
Backport of #21143

### CODE REVIEW

**Comment by danxuliu** [comment]
Test failures are unrelated.

However, in a previous Drone run something strange happened in the `sharing-v1` integration test. _Creating a new mail share_ failed because the mailer tried to connect to port 25 instead of to port 2525 as configured. I guess that the OCC call failed, but I have no idea why; also the rest of integration tests worked fine, locally it works and running again in Drone worked too :shrug:

**Comment by danxuliu** [comment]
As now it is no longer allowed to disable _password protect by Talk_ without setting a new password the UI had to be adjusted. The change is specific to this backport, as #20938 will be backported only to 19 and 18.

**Review by rullzer — APPROVED** [review]

File: build/integration/features/bootstrap/FakeSMTPHelper.php [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		$this->mail['emailRecipients'] = [];
+		$this->mail['emailSubject'] = false;
+		$this->mail['rawEmail'] = false;
+		$this->mail['emailHeaders'] = false;
+		$this->mail['emailBody'] = false;
+
+		$this->fd = $fd;
+	}
+
+	public function receive() {
```
Comment (ChristophWurst) [inline build/integration/features/bootstrap/FakeSMTPHelper.php:29]: :eyes: what am I reading here  :open_mouth:

**Review by ChristophWurst — APPROVED** [review]
:elephant: 

Those tests :exploding_head:

**Pull request merged**